core/src/main/scala/it/scalachess/core/logic/moves/generators/GenerateBishopMoves.scala

```diff
@@ -0,0 +1,42 @@
+package it.scalachess.core.logic.moves.generators
+
+import it.scalachess.core.Color
+import it.scalachess.core.board.{ Board, Position }
+import it.scalachess.core.logic.moves.ValidMove
+import it.scalachess.core.pieces.Bishop
+
+private[generators] object GenerateBishopMoves extends GeneratePieceMoves {
+  override def apply(color: Color, board: Board, from: Position): List[ValidMove] =
+    generateLinearMovementSimpleMoves(Bishop,
+                                      color,
+                                      board,
+                                      from,
+                                      Position.of(from.col + from.colRightMod, from.row + from.rowUpMod),
+                                      from.colRightMod,
+                                      from.rowUpMod,
+                                      List()) ++
```

Comment:
List() should not be passed to a tailrec, its better to make another method.
This method takes the same parameters except the List() and calls the tailrec by passing the empty list